### PR TITLE
diary: headers, lists, and rules

### PR DESCRIPTION
**Pull request by arthyn**
This implements #952 and additionally horizontal rules. It does not add any of the formatting UI to the menu only the extension support itself. so you can type them in, but we still need the extended formatting menu for notebooks @nerveharp .

Also adds state migration for diary so we don't blow up any state.

### CODE REVIEW

**Comment by nerveharp** [comment]
Thank you, chef

**Comment by arthyn** [comment]
we gotta cook

File: desk/sur/diary.hoon [hunk 1]
```diff
@@ -120,14 +120,24 @@
...
 ::
 ::    %image: a visual, we record dimensions for better rendering
 ::    %cite: an Urbit reference
+::    %header: a traditional HTML heading, h1-h6
+::    %listing: a traditional HTML list, ul and ol
 ::
 +$  block
   $%  [%image src=cord height=@ud width=@ud alt=cord]
       [%cite =cite:c]
+      [%header p=?(%h1 %h2 %h3 %h4 %h5 %h6) q=inline]
```
Comment (liam-fitzgerald) [inline desk/sur/diary.hoon:138]: should be `q=(list inline)`

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
Needs moar recursion

File: desk/sur/diary.hoon (hunk 1 above)
Comment (arthyn) [reply desk/sur/diary.hoon:138]: bah ok fine

Comment (liam-fitzgerald) [reply desk/sur/diary.hoon:138]: come on..., it's like a 10 line change tops ;)

**Comment by arthyn** [comment]
got verbal approval

**Pull request merged**